compiler/vm.nim

```diff
@@ -567,6 +610,24 @@ proc rawExecute(c: PCtx, start: int, tos: PStackFrame): TFullReg =
       let rb = instr.regB
       ensureKind(rkFloat)
       regs[ra].floatVal = cast[float64](int64(regs[rb].intVal))
+    of opcCastPtrToInt:
+      let rb = instr.regB
+      ensureKind(rkInt)
+      if regs[rb].kind != rkNode:
+        stackTrace(c, tos, pc, "opcCastPtrToInt: regs[rb].kind: " & $regs[rb].kind)
+      regs[ra].intVal = cast[int](regs[rb].node.intVal)
+    of opcCastIntToPtr:
+      let rb = instr.regB
+      let typ = regs[ra].node.typ
```

Comment:
This is dangerous, keep in mind that ``n.typ == nil`` for PNode in the VM.

Reply:
I think the way i wrote is correct because vm codegen has `opcSetType`:
```
c.gABx(n, opcSetType, dest, c.genType(dst))
c.gABC(n, opcCastIntToPtr, dest, tmp)
```
I did extensive testing (see link in PR msg: `further examples`)